src/pt/yugenmangas/src/eu/kanade/tachiyomi/extension/pt/yugenmangas/YugenMangas.kt

```diff
@@ -49,102 +43,97 @@ class YugenMangas : ParsedHttpSource() {
     override fun headersBuilder(): Headers.Builder = Headers.Builder()
         .add("Referer", "$baseUrl/")
 
-    override fun popularMangaRequest(page: Int): Request = GET(baseUrl, headers)
+    val apiHeaders by lazy { apiHeadersBuilder().build() }
 
-    override fun popularMangaSelector(): String = "div.popular div.swiper-wrapper a"
+    private fun apiHeadersBuilder(): Headers.Builder = headersBuilder()
+        .add("Accept", "application/json, text/plain, */*")
+        .add("Origin", baseUrl)
+        .add("Sec-Fetch-Dest", "empty")
+        .add("Sec-Fetch-Mode", "cors")
+        .add("Sec-Fetch-Site", "same-site")
 
-    override fun popularMangaFromElement(element: Element): SManga = SManga.create().apply {
-        title = element.selectFirst("h1")!!.text()
-        thumbnail_url = element.selectFirst("img")!!.absUrl("src")
-        url = element.attr("href")
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$API_BASE_URL/random_top_series", apiHeaders)
```

Comment:
You can skip a 301 redirect:
```suggestion
        return GET("$API_BASE_URL/random_top_series/", apiHeaders)
```

Reply:
Was searching for these, thanks.